launchable/commands/split_subset.py

```diff
@@ -180,18 +211,44 @@ def run(self):
                         err=True)
                     return
 
-            if len(output) == 0:
-                click.echo(click.style("Error: no tests found in this subset id.", 'yellow'), err=True)
-                return
+            if is_split_by_groups:
+                group_names = []
+                for group in split_groups:
+                    group_name = group.get("groupName", "")
+                    subset = group.get("subset", [])
+                    rests = group.get("rest", [])
+
+                    if is_observation:
+                        subset, rests = subset + rests, []
+                    if is_output_exclusion_rules:
+                        subset, rests = rests, subset
+
+                    if len(subset) > 0:
+                        self.write_file("{}/subset-{}.txt".format(split_by_groups_output_dir, group_name), subset)
+                        if group_name != SPLIT_BY_GROUPS_NO_GROUP_NAME:
+                            group_names.append(group_name)
+
+                    if len(rests) > 0:
```

Comment:
Per spec, this should be produced only when --rest is used.

Reply:
click v8 supports optional value https://click.palletsprojects.com/en/8.1.x/options/. 
However, we use click v7 when the python version is under 3.6 https://github.com/launchableinc/cli/blob/main/setup.cfg#L18-L20

Hum... how to implement the rest option with the split-by-groups option ...